### ISSUE TITLE
Unable to Reproduce Results with DeBERTa-v3-base and Questions on Epochs

### ISSUE THREAD

**Issue by kaizhuanren** [yueyu1030/AttrPrompt#8]
Hello,

I'm facing a couple of issues while trying to replicate the results . I am using the DeBERTa-v3-base model and have followed the provided instructions closely. However, I encountered some problems that I hope you could help me resolve.

Question about Training Epochs:
I noticed in the code that only 6 epochs are used for training. Could you provide more context on why this number was chosen? Is it based on any specific observations or experiments?
Difficulty in Reproducing Results:
I attempted to reproduce the experimental results by using almost the same hyperparameters as specified, with the exception of the versions of the packages, CUDA, and the GPU model.
Despite these minor differences, I couldn't achieve the results reported in the paper. Notably, the performance is nearly identical when using both real and generated datasets, which seems unusual.
Additionally, even when training on the test set, the model does not overfit, which leads me to suspect there might be an issue with the codebase or my setup.
To ensure I've configured everything correctly, I double-checked the following:

The input and output paths are correctly set.
The rest of the hyperparameters match those in the original configuration.
I have tried different datasets (real and synthetic) but observed no significant difference in outcomes.
Could you please check if there's anything I may be missing or if there is a potential bug in the repository that could affect the reproducibility of the results? Any advice or guidance would be greatly appreciated.

Thank you in advance for your time and assistance.
The log is as bellow:
[test_tryoverfit_log_file.txt](https://github.com/user-attachments/files/17404873/test_tryoverfit_log_file.txt)
[train_with_Ture_data_log_file.txt](https://github.com/user-attachments/files/17404874/train_with_Ture_data_log_file.txt)

**Comment by yueyu1030** [comment]
Hi,

Thanks for reaching out. After reviewing your log, I noticed the following lines:

```
10/16/2024 19:30:33 - INFO - utils -   Loading features from cached file amazon/cache/cached_train_amazon_deberta-v3-base_128
10/16/2024 19:30:34 - INFO - utils -   Loading features from cached file amazon/cache/cached_dev_amazon_deberta-v3-base_128
10/16/2024 19:30:34 - INFO - utils -   Loading features from cached file amazon/cache/cached_test_amazon_deberta-v3-base_128
```
This suggests that the same cached files might be reused across different training rounds. To avoid this, try assigning unique names to your cached files for each dataset or training run.

**Comment by kaizhuanren** [comment]
Thank you for your prompt response and the insightful suggestion. I have indeed implemented the unique naming for the cached files, which has helped to resolve some of the issues I was facing.

I'm happy to say that I can now reproduce most of the results, but there's still one thing that's bugging me. During some of the experiments, the loss on the test set occasionally turns into NaN during evaluation. Have you ever run into something similar? Could it be due to data preprocessing, numerical instability, or maybe something else entirely?

Any thoughts or advice you could share would be super helpful.

**Comment by ritaranx** [comment]
I haven't encountered this issue in my experiments, but I recommend printing the model's predictions along with the labels. This way, you can check if any "NaN" values appear in either the test labels or the model's output.

**Comment by yueyu1030** [comment]
I haven't encountered this issue in my experiments, but I recommend printing the model's predictions along with the labels. This way, you can check if any "NaN" values appear in either the test labels or the model's output.

**Comment by kaizhuanren** [comment]
Thank you for your reply!
